taui/src/selectors/detail-listing.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import lonlat from '@conveyal/lonlat'
+import get from 'lodash/get'
+import {createSelector} from 'reselect'
+
+// import {NeighborhoodProperties} from '../types'
+import scale from '../utils/scaling'
+
+import selectListingRoute from './network-listing-route'
+import listingTravelTime from './listing-travel-times'
+
+// ~130km estimate of maximum straight-line distance drivable in area within 120 minutes
+const MAX_DISTANCE = 130000
+const MAX_TRAVEL_TIME = 120
+
+/* eslint complexity: 0 */
+export default createSelector(
+  selectListingRoute,
+  listingTravelTime,
+  state => get(state, 'data.activeListing'),
+  state => get(state, 'data.origin'),
+  state => get(state, 'data.userProfile'),
+  (ListingRoute, travelTime, listing, origin, profile) => {
+    if (!listing || !profile || !ListingRoute) {
+      return null
+    }
+    const useTransit = !profile || !profile.hasVehicle
+
+    const segments = useTransit ? ListingRoute.routeSegments : []
+    const time = useTransit ? travelTime : distanceTime(origin, listing)
+    // Lisitings called from within routable neighborhoods filtered within max travel time window,
+    // so it's assumed listings also within max travel time window.
+    // Smaller travel time is better; larger timeWeight is better (reverse range).
+    const timeWeight = time < MAX_TRAVEL_TIME ? scale(time, 0, MAX_TRAVEL_TIME, 1, 0) : 1
```

Comment:
Thanks for this catch! You’re right that routes that exceed the `MAX_TRAVEL_TIME` should ultimately be filtered out and `timeWeight` should be 0 if any slip through. Actually, I included this variable to try and keep route time creation/route state consistent between neighborhoods and listings, but looking at this more closely `timeWeight` is really only used to create the `score` in which `neighborhoodsSortedWithRoutes` are ordered. Since we don’t create a recommended order of listings in a neighborhood, I think we can just get rid of the `timeWieght` variable here.